architecture/2025-01-serverless-feature-spec.md

```diff
@@ -0,0 +1,534 @@
+# Topic: Serverless Container Platforms
+
+* **Author**: Will Tsai (@willtsai)
+
+## Topic Summary
+<!-- A paragraph or two to summarize the topic area. Just define it in summary form so we all know what it is. -->
+Given the importance of serverless infrastructure in the modern application landscape, it is a priority for Radius to expand beyond Kubernetes and support additional container platforms with lower operational overhead. The initial expansion will focus on support for Azure Container Instances, then AWS Elastic Container Service including AWS Fargate. This will be followed by more feature rich platforms including Azure Container Apps and eventually Google CloudRun.
```

Comment:
> The initial expansion will focus on support for Azure Container Instances, then AWS Elastic Container Service including AWS Fargate. This will be followed by more feature rich platforms including Azure Container Apps and eventually Google CloudRun.

Just curious - how did we decide the priority of different platforms?

Reply:
We wanted the first expansion beyond Kubernetes to be into an unopinionated compute platform that provides raw compute primitives that we can build upon. This would allow us to better learn how to model the extensibility of Radius core that paves the way for expansion into more opinionated or feature rich compute offerings.